extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -294,6 +347,72 @@ export class PublishDatabaseDialog {
 		return flexRadioButtonsModel;
 	}
 
+	private createPublishTypeRadioButtons(view: azdataType.ModelView): azdataType.Component {
+		const publishToLabel = view.modelBuilder.text().withProps({
+			value: 'Publish To'
+		}).component();
+		this.existingServerRadioButton = view.modelBuilder.radioButton()
+			.withProps({
+				name: 'publishType',
+				label: constants.publishToExistingServer
+			}).component();
+
+		this.existingServerRadioButton.checked = true;
+		this.existingServerRadioButton.onDidChangeCheckedState((checked) => {
+			this.onPublishTypeChange(checked);
+		});
+
+		this.dockerServerRadioButton = view.modelBuilder.radioButton()
+			.withProps({
+				name: 'publishType',
+				label: constants.publishToDockerContainer
+			}).component();
+
+		this.dockerServerRadioButton.onDidChangeCheckedState((checked) => {
+			this.onPublishTypeChange(!checked);
+		});
+
+		let flexRadioButtonsModel: azdataType.FlexContainer = view.modelBuilder.flexContainer()
+			.withLayout({ flexFlow: 'column' })
+			.withItems([publishToLabel, this.existingServerRadioButton, this.dockerServerRadioButton])
+			.withProps({ ariaRole: 'radiogroup' })
+			.component();
+
+		return flexRadioButtonsModel;
+	}
+
+	private onPublishTypeChange(existingServer: boolean) {
+		this.existingServerSelected = existingServer;
+		this.tryEnableGenerateScriptAndOkButtons();
+		if (existingServer) {
+			if (this.connectionRow) {
+				this.formBuilder!.insertFormItem({
+					title: '',
+					component: this.connectionRow
+				}, 2);
+			}
+			if (this.localDbSection) {
+				this.formBuilder!.removeFormItem({
+					title: '',
+					component: this.localDbSection
+				});
+			}
+		} else {
+			if (this.connectionRow) {
+				this.formBuilder!.removeFormItem({
+					title: '',
+					component: this.connectionRow
+				});
+			}
+			if (this.localDbSection) {
+				this.formBuilder!.insertFormItem({
```

Comment:
since the publish to docker option will create a new server and database, maybe the database dropdown should be an inputbox instead since there aren't any other databases to choose.